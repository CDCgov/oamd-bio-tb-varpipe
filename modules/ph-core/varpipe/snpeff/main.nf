process VARPIPE_SNPEFF {
    tag "$meta.id"
    label 'process_medium'

    container 'oamd-bio-snpeff:4.3r_a5da4a7_v1'

    input:
    tuple val(meta), path(vcf)
    val   db
    tuple val(meta2), path(cache)

    output:
    tuple val(meta), path("*.ann.vcf"),       emit: vcf
    tuple val(meta), path("*.csv"),       emit: report
    tuple val(meta), path("*.html"),      emit: summary_html
    tuple val(meta), path("*.genes.txt"), emit: genes_txt
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def avail_mem = 6144
    if (!task.memory) {
        log.info '[snpEff] Available memory not known - defaulting to 6GB. Specify process memory requirements to change this.'
    } else {
        avail_mem = (task.memory.mega*0.8).intValue()
    }
    def prefix = task.ext.prefix ?: "${meta.id}"
    def cache_command = cache ? "-dataDir \${PWD}/${cache}" : ""
    """
    zcat ${vcf} | sed 's/NC_000962.3/NC_000962/g' | gzip > tmp.vcf.gz

    java -jar /usr/src/snpEff/snpEff/snpEff.jar \\
        $db \\
        $args \\
        -csvStats ${prefix}.csv \\
        $cache_command \\
        tmp.vcf.gz \\
        > ${prefix}.ann.vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        snpeff: \$(echo \$(java -jar /usr/src/snpEff/snpEff/snpEff.jar -version 2>&1) | cut -f 2 -d ' ')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.ann.vcf
    touch ${prefix}.csv
    touch ${prefix}.html
    touch ${prefix}.genes.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        snpeff: \$(echo \$(java -jar /usr/src/snpEff/snpEff/snpEff.jar -version 2>&1) | cut -f 2 -d ' ')
    END_VERSIONS
    """

}
