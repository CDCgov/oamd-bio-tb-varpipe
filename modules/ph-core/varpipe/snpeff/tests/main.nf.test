nextflow_process {

    name "Test Process VARPIPE_SNPEFF"
    script "../main.nf"
    process "VARPIPE_SNPEFF"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/snpeff"

    test("VARPIPE_SNPEFF vcf") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/vcf/test.vcf.gz', checkIfExists: true) ]
                ]
                input[1] = "NC_000962"
                input[2] = [
                    [ id:'cache' ], // meta map
                    [ ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.vcf,
                    process.out.genes_txt,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("VARPIPE_SNPEFF vcf -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/vcf/test.vcf.gz', checkIfExists: true) ]
                ]
                input[1] = "NC_000962"
                input[2] = [
                    [ id:'cache' ], // meta map
                    [ ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
