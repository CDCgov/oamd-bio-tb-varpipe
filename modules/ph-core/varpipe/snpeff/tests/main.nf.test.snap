{
    "VARPIPE_SNPEFF vcf": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.ann.vcf:md5,29548aed80468e270f36753d686b5c93"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.genes.txt:md5,ff1958910bdccf60c4ef9d3edc4d77e6"
                ]
            ],
            [
                "versions.yml:md5,9c4c819c37488a08f516e9e6e10ae4c0"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-16T10:43:56.889702038"
    },
    "VARPIPE_SNPEFF vcf -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.ann.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.genes.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,9c4c819c37488a08f516e9e6e10ae4c0"
                ],
                "genes_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.genes.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "report": [
                    [
                        {
                            "id": "test"
                        },
                        "test.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "summary_html": [
                    [
                        {
                            "id": "test"
                        },
                        "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.ann.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9c4c819c37488a08f516e9e6e10ae4c0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-16T10:44:04.644306083"
    }
}
